### PR TITLE
[micromamba] Update to v1.2.0

### PR DESCRIPTION
**Pull request by tylerjthomas9**
Release notes: https://github.com/mamba-org/mamba/releases/tag/2023.01.16

In case you're interested @cjdoris

### CODE REVIEW

**Comment by cjdoris** [comment]
Looks great, thank you!

File: M/micromamba/build_tarballs.jl [hunk 1]
```diff
@@ -3,53 +3,49 @@
...
-        "e8b02c5b51eecb71953a460846f37acd6066eb3cef6cd3aebc60835048f37ad3",
+        "5e51be4efa4f0e421077d153cec52c0868863a9e3c28198b92afe1c2150d83d8",
         filename="micromamba-x86_64-w64-mingw32.tar.bz2"),
 ]
 
 # Bash recipe for building across all platforms
 script = raw"""
-echo target=$target
-
 # unpack the tarball (BinaryBuilder does not natively support bzip2 so we do this ourselves)
```
Comment (giordano) [inline M/micromamba/build_tarballs.jl:34]: I just noticed this, wut?!

**Comment by giordano** [comment]
Also, this is using the non-permanent URLs, isn't it?

File: M/micromamba/build_tarballs.jl (hunk 1 above)
Comment (cjdoris) [reply M/micromamba/build_tarballs.jl:34]: A relic from debugging the script :)

**Pull request merged**